src/stores/AppStore/AppStore.ts

```diff
@@ -603,6 +603,9 @@ export class AppStore {
                 this.setSpectralMatchingEnabled(newFrame, true);
             }
             if (this.preferenceStore.autoWCSMatching & WCSMatchingType.SPATIAL && this.spatialReference !== newFrame) {
+                if (this.overlayStore.isImgCoordinates) {
+                    AST.setI(newFrame.wcsInfo, "Current", 1);
```

Comment:
I suggest we instead call `FrameStore.updateWcsSystem` in `FrameStore` `constructor` to make sure that `wcsInfo` is correctly updated after we create a frame.

Reply:
yes, that is more thoughtful!